index.js

```diff
@@ -1,125 +1,201 @@
 'use strict'
 
 const nodePath = require('path')
-
-const commonExtname = require('common-extname')
-const commonPathPrefix = require('common-path-prefix')
-const glob = require('glob')
+const GlobSync = require('glob/sync').GlobSync
+const GLOBSTAR = require('minimatch').GLOBSTAR
+const hasMagic = require('glob').hasMagic
 const identifierfy = require('identifierfy')
-const flatten = require('lodash.flatten')
 
-function getPattern (path, source) {
-  const pattern = source.value.replace(/^glob:/, '').trim()
-  if (!pattern) {
-    throw path.buildCodeFrameError(`Missing glob pattern '${source.value}'`)
+const dirname = nodePath.dirname
+const relative = nodePath.relative
+const resolve = nodePath.resolve
+const fileSeparator = nodePath.sep
+
+function makeRe (set) {
+  let min, max
+  const parts = []
+  for (let i = 0; i < set.length; i++) {
+    const p = set[i]
+    if (typeof p !== 'string') {
+      if (p === GLOBSTAR) {
+        parts.push('(?:(?!(?:/|^)\\.).)*?')
+      } else {
+        parts.push(p._src)
+      }
+      if (min === undefined) {
+        min = i
+      }
+      max = i
+    } else {
+      parts.push(p.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&'))
```

Comment:
This escapes the literal string. I had this wrapped in a function, but I only used it once. I could make it a function again. Or comment it. I pulled it from minimatch.